### DISCUSSION TITLE
lora sx1278 encryption

### DISCUSSION THREAD

**Discussion by anandspathak** [jgromes/RadioLib#955] [Q&A]
Dear Sir,
Can You suggest me a library for encryption and decryption of lora sx1278 data packets, or ill be grateful if you can provide an example code for the same,
thanks,
Regards,
Anand Pathak
INDIA

**Comment by jgromes** [comment]
RadioLib contains AES128 encryption/decryption functions. It should be noted that currently, only ECB mode is supported. There is no specific example, but it's pretty easy to use - see https://jgromes.github.io/RadioLib/class_radio_lib_a_e_s128.html

```c++
uint8_t key[16]; // your secrect key
RadioLibAES128Instance.init(key);
RadioLibAES128Instance.encryptECB(inputData, dataLen, outputData);
```

**Reply by anandspathak** [reply]
thank you for your prompt reply,
I don't have any experience with encryption, all i needed to ensure 2 below points that:
1) Only pair with same encryption and decryption keys only talk to each other and none other( i am using 3 pairs in my house),
2) any outsider with their own lora should not get my actual contents of data packets....(its ok if they get to know that something is being transmitted)

does the aes library you suggested FULFILS MY NEED???
thanks again for your valuable reply,
Regards,
Anand Pathak,
INDIA

**Reply by HeadBoffin** [reply]
If you are concerned about security it doesn't seem appropriate to ask a stranger if some code will be OK, so you may want to do a little research yourself. And without anyone knowing what it is you are trying to keep secret and/or some commercial arrangement with liability insurance, you've no real recourse on any answer.

With the AES library that is available you will have excellent encryption but for LoRa to LoRa transmissions you will have to call the encryption & decryption routines from your code. You'll also have to setup the keys so that both ends can use the AES. It's deemed suitable by thousands of companies implementing LoRaWAN but you could review the Wikipedia page on AES to make a determination as to whether it will be strong enough for you.

Anyone listening with a LoRa radio of some sort may be able to see the packets which is technically the same as getting the actual contents, but without the keys won't be able to decrypt them so they won't know what it is about.

**Reply by anandspathak** [reply]
Thank you,
Regards to both,
Anand Pathak
